### PR TITLE
chore: enhance pipeline_perf arguments

### PR DESCRIPTION
**Pull request by andreitraistaru**
## Reason

Allow specifing multiple instance types and platforms after `--instances` and `--platforms` flags.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: .buildkite/pipeline_perf.py [hunk 1]
```diff
@@ -88,12 +88,13 @@ def build_group(test):
     default=[],
 )
 args = parser.parse_args()
-if not args.instances:
-    args.instances = DEFAULT_INSTANCES
 if not args.platforms:
     args.platforms = DEFAULT_PLATFORMS
+else:
+    args.platforms = [(str(platform).split("-")[0], str(platform).split("-")[1]) for platform in args.platforms]
```
[the remark this answers is not in the document]
Comment (pb8o) [reply .buildkite/pipeline_perf.py]: Also we have to be careful not to add `-` in the os or kernel version, but that is already the case.

[the remark this answers is not in the document]
Comment (andreitraistaru) [reply .buildkite/pipeline_perf.py]: Yeah, I had the same concern here. We can change it in `=` which has a lower change of being in the name of the os/kv. What do you think?

File: .buildkite/pipeline_perf.py [hunk 2]
```diff
@@ -88,12 +88,14 @@ def build_group(test):
...
-if not args.instances:
-    args.instances = DEFAULT_INSTANCES
 if not args.platforms:
     args.platforms = DEFAULT_PLATFORMS
+else:
+    args.platforms = [(str(platform).split("-", maxsplit=1)[0], str(platform).split("-", maxsplit=1)[1])
+                      for platform in args.platforms]
 if args.extra:
     args.extra = dict(val.split("=", maxsplit=1) for val in args.extra)
+print(args)
```
Comment (andreitraistaru) [inline .buildkite/pipeline_perf.py:98]: This is a leftover. I'll remove it now

File: .buildkite/pipeline_perf.py [hunk 3]
```diff
@@ -70,14 +70,14 @@ def build_group(test):
 parser.add_argument(
     "--instances",
     required=False,
-    action="append",
```
[the remark this answers is not in the document]
Comment (andreitraistaru) [reply .buildkite/pipeline_perf.py:73]: Are you sure? I tested like this and works: 

```
python3 .buildkite/pipeline_perf.py --test block --platforms a-b --instances a
{
    "env": {
        "AWS_EMF_NAMESPACE": "PerfTests",
        "AWS_EMF_SERVICE_NAME": "PerfTests"
    },
    "steps": [
        {
            "group": "🖴 Block Performance",
            "steps": [
                {
                    "agents": [
                        "instance=a",
                        "kv=b",
                        "os=a",
                        "ag=1"
                    ],
                    "artifacts": [
                        "./test_results/*"
                    ],
                    "command": [
                        "./tools/devtool -y test -r 16834m -c 1-10 -m 0 -- --nonci -s --dump-results-to-file --log-cli-level=INFO integration_tests/performance/test_block_performance.py"
                    ],
                    "label": "🖴 a a b",
                    "timeout_in_minutes": 240
                }
            ]
        }
    ]
```
`nargs='+'` doesn't allow zero args (I mean, it doesn;t allow something like this `python3 .buildkite/pipeline_perf.py --test block --instances`) but in that case there is no need to add `--instances` we can just let it `python3 .buildkite/pipeline_perf.py --test block`

File: .buildkite/pipeline_perf.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pb8o) [reply .buildkite/pipeline_perf.py]: I think `-` is fine here. We already make the assumption that there's no `-` in the os or kernel or any fields for that matter.

File: .buildkite/pipeline_perf.py [hunk 4]
```diff
@@ -88,10 +88,11 @@ def build_group(test):
     default=[],
 )
 args = parser.parse_args()
-if not args.instances:
-    args.instances = DEFAULT_INSTANCES
 if not args.platforms:
     args.platforms = DEFAULT_PLATFORMS
+else:
+    args.platforms = [(str(platform).split("-", maxsplit=1)[0], str(platform).split("-", maxsplit=1)[1])
```
Comment (pb8o) [inline .buildkite/pipeline_perf.py:94]: I think you only need one split:

```
    args.platforms = [tuple(platform.split("-", maxsplit=1)) ...
```

**Review by pb8o — APPROVED** [review]

File: .buildkite/pipeline_perf.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pb8o) [reply .buildkite/pipeline_perf.py:73]: Ah ok, I misunderstood `append` with a default value then.

**Review by pb8o — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Review by roypat — APPROVED** [review]

**Pull request merged**